Config form notes interface (#232)
mautic/developer-documentation-new: docs/components/integrations.rst

1. :ref:`Register the Integration<components/integrations_builder:Register the Integration as a builder>` as an Integration that provides a custom builder.
2. Configure what featured builders the Integration supports (Mautic currently supports 'Email' and 'Landing Page' builders).

.. vale off

> Config Form Notes Interface

Reviewer: We're using sentence case for titles. So, unless necessary, only the first letter is capitalized.

```suggestion
Config Form notes interface
```

***************************

.. vale on

> The interface, :code:`\Mautic\IntegrationsBundle\Integration\Interfaces\ConfigFormNotesInterface`, provides a way to add notes, either info or warning, to the Plugin configuration form.

Reviewer: Let's do dash instead of comma here for readability.

```suggestion
The interface, :code:`\Mautic\IntegrationsBundle\Integration\Interfaces\ConfigFormNotesInterface`, provides a way to add notes - either info or warning - to the Plugin configuration form.
```

> The interface, :code:`\Mautic\IntegrationsBundle\Integration\Interfaces\ConfigFormNotesInterface`, provides a way to add notes - either info or warning - to the Plugin configuration form.

Reviewer: **question**: For the code below,

```rst
:code:`\Mautic\IntegrationsBundle\Integration\Interfaces\ConfigFormNotesInterface`
```

Do you want this to render a particular code-block or inline code as the screenshot?

If it's inline code, we can just use double backtick as below:

```rst
``\Mautic\IntegrationsBundle\Integration\Interfaces\ConfigFormNotesInterface``
```

If it's a code block, please use `.. code-block::` and provide the code block.

### Screenshot

<img width="1431" height="317" alt="Image" src="https://github.com/user-attachments/assets/b31fed9d-4161-43e1-9fb0-b7ca340a5dac" />

> Learn more about it on :ref:`Integration Configuration Form Notes<components/integrations_configuration_form_notes:Integration configuration form notes>`.

Reviewer: Vale still gives warning. Let's address this by:

- Fix capitalization in all lines of code, particularly: `Plugin configuration Form` and `Integration configuration Form notes`.

- Wrap these lines in `.. vale off` and `.. vale on` because the other warnings aren't applicable here.

```suggestion
.. vale off

The interface, :code:`\Mautic\IntegrationsBundle\Integration\Interfaces\ConfigFormNotesInterface`, provides a way to add notes - either info or warning - to the Plugin configuration Form.

Learn more about it on :ref:`Integration configuration Form notes<components/integrations_configuration_form_notes:Integration configuration form notes>`.

.. vale on
```

> Read more about to how-tos :ref:`Integration Configuration Form Notes<components/integrations_configuration_form_notes:Integration configuration form notes>`

Reviewer: Small tweak here for better read.

```suggestion
Learn more about it on :ref:`Integration Configuration Form Notes<components/integrations_configuration_form_notes:Integration configuration form notes>`.
```

1. :ref:`Register the Integration<components/integrations_builder:Register the Integration as a builder>` as an Integration that provides a custom builder.
2. Configure what featured builders the Integration supports (Mautic currently supports 'Email' and 'Landing Page' builders).

.. vale off

Config Form Notes Interface
***************************

.. vale on

> The interface, :code:`\Mautic\IntegrationsBundle\Integration\Interfaces\ConfigFormNotesInterface`, provides a way to put notes, either info or warning, on the plugin configuration form.

Author: ```suggestion
The interface, :code:`\Mautic\IntegrationsBundle\Integration\Interfaces\ConfigFormNotesInterface`, provides a way to add notes, either info or warning, to the Plugin configuration form.
```
